docs/using-the-compiler.rst

```diff
@@ -0,0 +1,262 @@
+  --abi                 ABI specification of the contracts.
+  --interface           Solidity interface of the contracts.
+  --hashes              Function signature hashes of the contracts.
+  --userdoc             Natspec user documentation of all contracts.
+  --devdoc              Natspec developer documentation of all contracts.
+  --formal              Translated source suitable for formal analysis.
+
+          // to be defined
+        }
+      }
+    }
+
+
+Regular Output
+--------------
+
+
+    {
+      errors: ["error1", "error2"], // we might structure them
+      errors: [
+          {
+              // (axic)
+              file: "sourceFile.sol", // optional?
+              contract: "contractName", // optional
+              line: 100, // optional - currently, we always have a byte range in the source file
+              // Errors/warnings originate in several components, most of them are not
+              // backend-specific. Currently, why3 errors are part of the why3 output.
+              // I think it is better to put code-generator-specific errors into the code-generator output
+              // area, and warnings and errors that are code-generator-agnostic into this general area,
+              // so that it is easier to determine whether some source code is invalid or only
+              // triggers errors/warnings in some backend that might only implement some part of solidity.
+              type: "evm" or "why3" or "ewasm" // maybe a better field name would be needed
+              severity: "warning" or "error" // mandatory
+              message: "Invalid keyword" // mandatory
+          }
+      ]
+      contracts: {
+        "sourceFile.sol:ContractName": {
+          // The Ethereum Contract ABI. If empty, it is represented as an empty array.
+          // See https://github.com/ethereum/wiki/wiki/Ethereum-Contract-ABI
+          abi: [],
+          evm: {
+              assembly:
+              bytecode:
+              runtimeBytecode:
+              opcodes:
+              annotatedOpcodes: // (axic) see https://github.com/ethereum/solidity/issues/1178
+              gasEstimates:
+              sourceMap:
+              runtimeSourceMap:
+              // If given, this is an unlinked object (cannot be filtered out explicitly, might be
+              // filtered if both bytecode, runtimeBytecode, opcodes and others are filtered out)
+              linkReferences: {
+                "sourceFile.sol:Library1": [1, 200, 80] // byte offsets into bytecode. Linking replaces the 20 bytes there.
+              }
+              // the same for runtimeBytecode - I'm not sure it is a good idea to allow to link libraries differently for the runtime bytecode.
+              // furthermore, runtime bytecode is always a substring of the bytecode anyway.
+              runtimeLinkReferences: {
+              }
+          },
+          functionHashes:
+          metadata: // see below
```

Comment:
Mention that, contrary to the rest of the fields, this is not a JSON object, but a stringified JSON. This is because serialisation differences might result in different output and thus metadata hash.